core/src/main/java/com/linecorp/armeria/client/ClientFactoryBuilder.java

```diff
@@ -512,7 +516,17 @@ public ClientFactoryBuilder meterRegistry(MeterRegistry meterRegistry) {
      * The {@link ProxyConfig} which contains proxy related configuration.
      */
     public ClientFactoryBuilder proxyConfig(ProxyConfig proxyConfig) {
-        option(ClientFactoryOption.PROXY_CONFIG, proxyConfig);
+        requireNonNull(proxyConfig, "proxyConfig");
+        option(ClientFactoryOption.PROXY_CONFIG_SELECTOR, StaticProxyConfigSelector.of(proxyConfig));
+        return this;
+    }
+
+    /**
+     * The {@link ProxyConfigSelector} which contains proxy related configuration.
+     */
+    public ClientFactoryBuilder proxyConfigSelector(ProxyConfigSelector proxyConfigSelector) {
```

Comment:
- Could we also have an overloaded version that accepts `ProxySelector`, if `WrappingProxyConfigSelector` works with any custom `ProxySelector` implementations?
- How about renaming this method to `proxyConfig()`? Because:
  - It's shorter.
  - It seems to me that there won't be any conflict between any `proxyConfig()` methods we may add in the future.